osu.Game.Rulesets.Taiko/Replays/TaikoAutoGenerator.cs

```diff
@@ -113,11 +113,16 @@ public override Replay Generate()
                 else
                     throw new InvalidOperationException("Unknown hit object type.");
 
-                Frames.Add(new TaikoReplayFrame(endTime + KEY_UP_DELAY));
+                TaikoHitObject nextHitObject = i < Beatmap.HitObjects.Count - 1 ? Beatmap.HitObjects[i + 1] : null;
 
-                if (i < Beatmap.HitObjects.Count - 1)
+                bool canDelayKeyUp = nextHitObject == null || nextHitObject.StartTime > endTime + KEY_UP_DELAY;
+
+                if (canDelayKeyUp)
```

Comment:
It seems weird that a key-up frame isn't added if it can't delay. Why isn't this implemented similarly to mania's generator?

Reply:
Because it seemed to be working correctly, but I agree that it's weird and I'm already working on this.